Restructure influxdb3 /plugins for step-by-step clarity (#5963)
influxdata/docs-v2: content/shared/extended-plugin-api.md

        """Build the line protocol string."""
        # Start with measurement name (escape commas only)
        line = self.measurement.replace(',', '\\,')

        # Add tags if present
        if self.tags:
            tags_str = ','.join(
                f"{k}={v}" for k, v in self.tags.items()
            )
            line += f",{tags_str}"

        # Add fields (required)
        if not self.fields:
            raise InvalidLineError(f"At least one field is required: {line}")

        fields_str = ','.join(
            f"{k}={v}" for k, v in self.fields.items()
        )
        line += f" {fields_str}"

        # Add timestamp if present
        if self._timestamp_ns is not None:
            line += f" {self._timestamp_ns}"

        return line
```
{{% /expand %}}
{{% /expand-wrapper %}}

#### Query data

Your plugins can execute SQL queries and process results directly:

```python
# Simple query
results = influxdb3_local.query("SELECT * FROM metrics WHERE time > now() - INTERVAL '1 hour'")

# Parameterized query for safer execution
params = {"table": "metrics", "threshold": 90}
results = influxdb3_local.query("SELECT * FROM $table WHERE value > $threshold", params)
```

> Query results come back as a `List` of `Dict[String, Any]`, where each dictionary represents a row with column names as keys and column values as values. This makes it easy to process the results in your plugin code.

Reviewer: ```suggestion
Query results are a `List` of `Dict[String, Any]`, where each dictionary represents a row with column names as keys and column values as values.
```

We usually want to avoid making claims (for example, "makes it easy"). This one's not bad, but I would replace it with a brief example or just state that you can use standard Python to process the results.

#### Log information

The shared API `info`, `warn`, and `error` functions accept multiple arguments,
convert them to strings, and log them as a space-separated message to the database log.

Add logging to track plugin execution:

```python
influxdb3_local.info("Starting data processing")
influxdb3_local.warn("Could not process some records")
influxdb3_local.error("Failed to connect to external API")

# Log structured data
obj_to_log = {"records": 157, "errors": 3}
influxdb3_local.info("Processing complete", obj_to_log)
```

> All log messages appear in the server logs and are stored in system tables that you can query using SQL.

Reviewer: Link to the System table information

> #### Maintaining state with the in-memory cache

Reviewer: ```suggestion
#### Maintain state with the in-memory cache
```

The Processing engine provides an in-memory cache system that enables plugins to persist and retrieve data between executions.

You can access the cache through the `cache` property of the shared API:

```python 
# Basic usage pattern  
influxdb3_local.cache.METHOD(PARAMETERS)
```

> Here are the available methods: 

Reviewer: ```suggestion

`cache` provides the following methods to retrieve and manage cached values: 
```

```python
# Cache external API responses for 5 minutes  
influxdb3_local.cache.put("weather_data", api_response, ttl=300)
```

##### Cache computation results

Store the results of expensive calculations that need to be utilized frequently:

```python
# Cache aggregated statistics  
influxdb3_local.cache.put("daily_stats", calculate_statistics(data), ttl=3600)
```

##### Warm the cache

For critical data, prime the cache at startup. This can be especially useful for global namespace data where multiple triggers need the data:

```python
# Check if cache needs to be initialized  
if not influxdb3_local.cache.get("lookup_table"):   
    influxdb3_local.cache.put("lookup_table", load_lookup_data())
```

##### Consider cache limitations

- **Memory Usage**: Since cache contents are stored in memory, monitor your memory usage when caching large datasets.
- **Server Restarts**: Because the cache is cleared when the server restarts, design your plugins to handle cache initialization (as noted above).
- **Concurrency**: Be cautious of accessing inaccurate or out-of-date data when multiple trigger instances might simultaneously update the same cache key.

### Next Steps

> Now that you understand the Extended Plugin API, you're ready to build data processing workflows that can transform, analyze, and respond to your time series data.
> 
> Try combining these features to create plugins that:
> 
> - Process incoming data and write transformed results
> - Maintain state between executions using the cache
> - Log detailed information about their operation
> - Share configuration through the global cache

Reviewer: ```suggestion
With an understanding of the InfluxDB 3 Shared Plugin API, you're ready to build data processing workflows that can transform, analyze, and respond to your time series data or extend example plugins from the [plugin repo]() on GitHub.
```

Is "Extended Plugin API" the real name? It was the "Shared API" (which isn't a great name). Make sure to use the correct name consistently.
